### ISSUE TITLE
The EditableVectorLayer hangs after I updated to 4.1.3 RC1

### ISSUE THREAD

**Issue by Shtong** [CartoDB/mobile-sdk#214]
Hello,

I want to experiment with `EditableVectorLayer` to prepare our incoming features. For this I used a small test project (targeting Xamarin Android) that only contains a map, and copied the code from the `VectorObjectEditingActivity` class in the `mobile-dotnet-samples` project (as well as the required event listener classes and extension methods) to fill the map with some editable shapes. This project was already configured to use version 4.1.3 RC1 of the Carto library.

If I launch the project with carto 4.1.3 RC1: A few seconds after I start interacting with the map, the application completely hangs. Eventually the system popup for unresponsive app appears, asking me if i want to wait or kill my application.

Here are the logs of my application, starting when the map initially appears (before I start dragging the map), and stopping after the unresponsive popup has appeared:

```
04-25 14:58:01.998 D/ViewRootImpl@6e84ebe[MainActivity](25556): MSG_RESIZED_REPORT: frame=Rect(0, 0 - 1280, 800) ci=Rect(0, 32 - 0, 0) vi=Rect(0, 32 - 0, 0) or=2
04-25 14:58:01.998 D/ViewRootImpl@6e84ebe[MainActivity](25556): MSG_WINDOW_FOCUS_CHANGED 1
04-25 14:58:01.998 D/ViewRootImpl@6e84ebe[MainActivity](25556): mHardwareRenderer.initializeIfNeeded()#2 mSurface={isValid=true -534009856}
04-25 14:58:01.999 V/InputMethodManager(25556): Starting input: tba=android.view.inputmethod.EditorInfo@1999134 nm : ReproNewCartoProject.ReproNewCartoProject ic=null
04-25 14:58:01.999 I/InputMethodManager(25556): startInputInner - mService.startInputOrWindowGainedFocus
04-25 14:58:02.011 D/InputTransport(25556): Input channel constructed: fd=80
04-25 14:58:02.018 V/InputMethodManager(25556): Starting input: tba=android.view.inputmethod.EditorInfo@ad8db5d nm : ReproNewCartoProject.ReproNewCartoProject ic=null
04-25 14:58:48.315 D/ViewRootImpl@6e84ebe[MainActivity](25556): ViewPostImeInputStage processPointer 0
04-25 14:58:48.317 E/BoostFramework(25556): BoostFramework() : Exception_1 = java.lang.ClassNotFoundException: Didn't find class "com.qualcomm.qti.Performance" on path: DexPathList[[],nativeLibraryDirectories=[/system/lib, /vendor/lib]]
04-25 14:58:48.328 D/Mono    (25556): DllImport searching in: '__Internal' ('(null)').
04-25 14:58:48.328 D/Mono    (25556): Searching for 'java_interop_jnienv_call_nonvirtual_boolean_method_a'.
04-25 14:58:48.328 D/Mono    (25556): Probing 'java_interop_jnienv_call_nonvirtual_boolean_method_a'.
04-25 14:58:48.328 D/Mono    (25556): Found as 'java_interop_jnienv_call_nonvirtual_boolean_method_a'.
04-25 14:58:48.332 D/Mono    (25556): DllImport searching in: '__Internal' ('(null)').
04-25 14:58:48.332 D/Mono    (25556): Searching for 'java_interop_jnienv_call_nonvirtual_int_method_a'.
04-25 14:58:48.332 D/Mono    (25556): Probing 'java_interop_jnienv_call_nonvirtual_int_method_a'.
04-25 14:58:48.332 D/Mono    (25556): Found as 'java_interop_jnienv_call_nonvirtual_int_method_a'.
04-25 14:58:48.337 D/Mono    (25556): DllImport searching in: '__Internal' ('(null)').
04-25 14:58:48.337 D/Mono    (25556): Searching for 'java_interop_jnienv_call_nonvirtual_float_method_a'.
04-25 14:58:48.337 D/Mono    (25556): Probing 'java_interop_jnienv_call_nonvirtual_float_method_a'.
04-25 14:58:48.337 D/Mono    (25556): Found as 'java_interop_jnienv_call_nonvirtual_float_method_a'.
04-25 14:58:48.340 D/Mono    (25556): DllImport searching in: 'carto_mobile_sdk' ('libcarto_mobile_sdk.so').
04-25 14:58:48.340 D/Mono    (25556): Searching for 'CSharp_BaseMapView_OnInputEvent'.
04-25 14:58:48.340 D/Mono    (25556): Probing 'CSharp_BaseMapView_OnInputEvent'.
04-25 14:58:48.340 D/Mono    (25556): Found as 'CSharp_BaseMapView_OnInputEvent'.
04-25 15:00:08.227 I/art     (25556): Thread[3,tid=25562,WaitingInMainSignalCatcherLoop,Thread*=0xde751c00,peer=0x12c205e0,"Signal Catcher"]: reacting to signal 3
04-25 15:00:08.228 I/art     (25556): 
04-25 15:00:08.356 I/art     (25556): Wrote stack traces to '/data/anr/traces.txt'
```

I then decided to downgrade the Carto API to the latest stable (4.1.2). After that, I was able to use the application and edit shapes as expected.

It should also be noted that I previously tested the new text features (discussed in #197 and #193) with the same project, using 4.1.3 RC1, without encountering this problem. The problem appeared after I added the editable layer.

**Comment by jaakla** [comment]
@mtehver please check

**Comment by mtehver** [comment]
Thanks for reporting this. Could you share your project? EditableVectorLayer related changes in SDK 4.1.3RC1 vs 4.1.2 are quite minimal, so this could be triggered by something else.

**Comment by Shtong** [comment]
Here is my test project:
[ReproNewCartoProject.zip](https://github.com/CartoDB/mobile-sdk/files/1951679/ReproNewCartoProject.zip)

**Comment by mtehver** [comment]
Thanks for the test! This is fixed now and will be in 4.1.3 release.
